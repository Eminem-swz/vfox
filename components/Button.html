<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Button/ButtonGroup 按钮 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.133cf44e.css">
    <link rel="modulepreload" href="/vfox/assets/Home.aecaacb9.js">
    <link rel="modulepreload" href="/vfox/assets/app.6b4e8d9e.js">
    <link rel="modulepreload" href="/vfox/assets/components_Button.md.0cc21628.lean.js">
    
    <meta name="twitter:title" content="Button/ButtonGroup 按钮 | Vfox for Vue3">
  <meta property="og:title" content="Button/ButtonGroup 按钮 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-3eed0030><div class="sidebar-button" data-v-3eed0030><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-3eed0030 data-v-d91ad9e0><!----> Vfox for Vue3</a><div class="flex-grow" data-v-3eed0030></div><div class="nav" data-v-3eed0030><nav class="nav-links" data-v-3eed0030 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item isExternal" href="https://godxiaoji.github.io/vfox/playground/" target="_blank" rel="noopener noreferrer" data-v-a00c0156>Playground <svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-a00c0156><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-1314d532><nav class="nav-links nav" data-v-1314d532 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item isExternal" href="https://godxiaoji.github.io/vfox/playground/" target="_blank" rel="noopener noreferrer" data-v-a00c0156>Playground <svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-a00c0156><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></a></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-1314d532><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Button">Button</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#button-props">Button Props</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#sizetype-的合法值">SizeType 的合法值</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#statetype-的合法值">StateType 的合法值</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#buttonpattern-的合法值">ButtonPattern 的合法值</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#buttonshape-的合法值">ButtonShape 的合法值</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#form-type-的合法值">form-type 的合法值</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#button-slots">Button Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#默认（-default）">默认（#default）</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#buttongroup-props">ButtonGroup Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#buttongroup-slots">ButtonGroup Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#默认（-default）-1">默认（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-33252f1f><div class="container" data-v-33252f1f><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="" src="https://godxiaoji.github.io/vfox/demo/#/Button" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-33252f1f><div><h1 id="button-buttongroup-按钮" tabindex="-1">Button/ButtonGroup 按钮 <a class="header-anchor" href="#button-buttongroup-按钮" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-JavaScript"><pre><code><span class="token keyword">import</span> <span class="token punctuation">{</span> Button<span class="token punctuation">,</span> ButtonGroup <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">&#39;vfox&#39;</span>
</code></pre></div><p>具体的引入方式可以参考<a href="./../guide/import.html">引入组件</a>。</p><h2 id="button-props" tabindex="-1">Button Props <a class="header-anchor" href="#button-props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>size</td><td>SizeType</td><td>&#39;large&#39;</td><td>否</td><td>按钮尺寸</td></tr><tr><td>type</td><td>StateType</td><td>&#39;default&#39;</td><td>否</td><td>按钮类型</td></tr><tr><td>pattern</td><td>ButtonPattern</td><td>&#39;default&#39;</td><td>否</td><td>按钮款式</td></tr><tr><td>shape</td><td>ButtonShape</td><td>&#39;rectangle&#39;</td><td>否</td><td>按钮形状</td></tr><tr><td>color</td><td>string</td><td></td><td>否</td><td>自定义色彩，支持 hex rgb hsl 等写法，详细效果<a href="./../design/color.html#自定义色彩">查看</a></td></tr><tr><td>ghost</td><td>boolean</td><td>false</td><td>否</td><td>是否使用幽灵按钮，幽灵按钮将按钮的内容反色，背景变为透明</td></tr><tr><td>disabled</td><td>boolean</td><td>false</td><td>否</td><td>是否禁用</td></tr><tr><td>icon</td><td>string | Component</td><td></td><td>否</td><td>图标，使用 <a href="./Icon.html">Icon</a> 组件</td></tr><tr><td>loading</td><td>boolean</td><td>false</td><td>否</td><td>名称前是否带 loading 图标，优先级大于 icon</td></tr><tr><td>form-type</td><td>string</td><td></td><td>否</td><td>&#39;submit&#39; / &#39;reset&#39;</td></tr></tbody></table><h3 id="sizetype-的合法值" tabindex="-1">SizeType 的合法值 <a class="header-anchor" href="#sizetype-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>large</td><td>默认大小，高度 48px，文字 17px，图标 21px</td></tr><tr><td>small</td><td>小尺寸，高度 28px，文字 12px，图标 18px</td></tr></tbody></table><h3 id="statetype-的合法值" tabindex="-1">StateType 的合法值 <a class="header-anchor" href="#statetype-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>default</td><td>次要</td></tr><tr><td>primary</td><td>主要</td></tr><tr><td>success</td><td>成功</td></tr><tr><td>warning</td><td>警告</td></tr><tr><td>danger</td><td>危险</td></tr></tbody></table><h3 id="buttonpattern-的合法值" tabindex="-1">ButtonPattern 的合法值 <a class="header-anchor" href="#buttonpattern-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>default</td><td>默认按钮</td></tr><tr><td>solid</td><td>细线按钮</td></tr><tr><td>dashed</td><td>虚线按钮</td></tr><tr><td>borderless</td><td>无边框按钮</td></tr><tr><td>gradient</td><td>渐变色按钮</td></tr></tbody></table><h3 id="buttonshape-的合法值" tabindex="-1">ButtonShape 的合法值 <a class="header-anchor" href="#buttonshape-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>rectangle</td><td>长方形按钮，小圆角</td></tr><tr><td>round</td><td>半圆角按钮</td></tr><tr><td>circle</td><td>圆圈按钮，搭配图标使用，不显示文字</td></tr><tr><td>square</td><td>正方形按钮，小圆角，搭配图标使用，不显示文字</td></tr></tbody></table><h3 id="form-type-的合法值" tabindex="-1">form-type 的合法值 <a class="header-anchor" href="#form-type-的合法值" aria-hidden="true">#</a></h3><table><thead><tr><th>值</th><th>说明</th></tr></thead><tbody><tr><td>submit</td><td>提交表单</td></tr><tr><td>reset</td><td>重置表单</td></tr></tbody></table><h2 id="button-slots" tabindex="-1">Button Slots <a class="header-anchor" href="#button-slots" aria-hidden="true">#</a></h2><h3 id="默认（-default）" tabindex="-1">默认（#default） <a class="header-anchor" href="#默认（-default）" aria-hidden="true">#</a></h3><div class="language-Vue"><pre><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-button</span> <span class="token attr-name">form-type</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>submit<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span>提交<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-button</span><span class="token punctuation">&gt;</span></span>
</code></pre></div><h2 id="buttongroup-props" tabindex="-1">ButtonGroup Props <a class="header-anchor" href="#buttongroup-props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>size</td><td>SizeType</td><td>&#39;large&#39;</td><td>否</td><td>每个按钮尺寸</td></tr><tr><td>pattern</td><td>ButtonPattern</td><td>&#39;default&#39;</td><td>否</td><td>每个按钮款式</td></tr><tr><td>shape</td><td>ButtonShape</td><td>&#39;rectangle&#39;</td><td>否</td><td>每个按钮形状</td></tr></tbody></table><h2 id="buttongroup-slots" tabindex="-1">ButtonGroup Slots <a class="header-anchor" href="#buttongroup-slots" aria-hidden="true">#</a></h2><h3 id="默认（-default）-1" tabindex="-1">默认（#default） <a class="header-anchor" href="#默认（-default）-1" aria-hidden="true">#</a></h3><div class="language-Vue"><pre><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-button-group</span> <span class="token attr-name">shape</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>square<span class="token punctuation">&quot;</span></span> <span class="token attr-name">size</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>small<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-button</span> <span class="token attr-name">type</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>default<span class="token punctuation">&quot;</span></span> <span class="token attr-name">icon</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>LeftOutlined<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-button</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-button</span> <span class="token attr-name">type</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>default<span class="token punctuation">&quot;</span></span> <span class="token attr-name">icon</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>HomeOutlined<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-button</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-button-group</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div></div><footer class="page-footer" data-v-33252f1f data-v-c772b490><div class="edit" data-v-c772b490><div class="edit-link" data-v-c772b490 data-v-11add020><!----></div></div><div class="updated" data-v-c772b490><!----></div></footer><div class="next-and-prev-link" data-v-33252f1f data-v-6889d424><div class="container" data-v-6889d424><div class="prev" data-v-6889d424><!----></div><div class="next" data-v-6889d424><a class="link" href="/vfox/components/Icon" data-v-6889d424><span class="text" data-v-6889d424>Icon</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-6889d424><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"components_actionsheet.md\":\"ffb0dfa3\",\"components_activityindicator.md\":\"65931797\",\"components_avatar.md\":\"2f20da4c\",\"components_backtop.md\":\"4c88b650\",\"components_badge.md\":\"5df16f00\",\"components_button.md\":\"0cc21628\",\"components_calendar.md\":\"088bd5fe\",\"components_cascader.md\":\"acfc8be2\",\"components_cell.md\":\"ae2d4356\",\"components_checkbox.md\":\"49f9ebdd\",\"components_circleprogress.md\":\"6876ffe7\",\"components_collapse.md\":\"07f7f963\",\"components_copy.md\":\"2df9d39a\",\"components_countdown.md\":\"24e902ef\",\"components_countup.md\":\"7975683a\",\"components_datepicker.md\":\"c711985b\",\"components_dialog.md\":\"2269b0ab\",\"components_divider.md\":\"0b81934d\",\"components_drawer.md\":\"7be6b455\",\"components_dropdown.md\":\"8b4c4f36\",\"components_empty.md\":\"75aee9fe\",\"components_fixed.md\":\"bef29c3b\",\"components_flatlist.md\":\"9973fcea\",\"components_form.md\":\"24e39c55\",\"components_group.md\":\"b17bcffb\",\"components_icon.md\":\"9f2c94e8\",\"components_image.md\":\"253cdd34\",\"components_imagepreview.md\":\"ed0cc731\",\"components_imageuploader.md\":\"838691fb\",\"components_indexview.md\":\"f0d76783\",\"components_input.md\":\"5c906814\",\"components_layout.md\":\"c05f3d2a\",\"components_loadmore.md\":\"4b3fe5bc\",\"components_modal.md\":\"15bac724\",\"components_navbar.md\":\"ce7e4829\",\"components_noticebar.md\":\"7c69a1c4\",\"components_notify.md\":\"824e3307\",\"components_numberkeyboard.md\":\"dd1ecf4c\",\"components_order.md\":\"db6831f7\",\"components_pagination.md\":\"23855aae\",\"components_picker.md\":\"f7ce9c6b\",\"components_popdialog.md\":\"a731ea2d\",\"components_popmenu.md\":\"1ead1d18\",\"components_popover.md\":\"ebdf71a2\",\"components_price.md\":\"5beab4e1\",\"components_progress.md\":\"00440d9d\",\"components_radio.md\":\"b181d010\",\"components_range.md\":\"471d2948\",\"components_rate.md\":\"c99cd936\",\"components_result.md\":\"99b89f3f\",\"components_scrolltab.md\":\"521cd023\",\"components_scrollview.md\":\"008c06a5\",\"components_searchbar.md\":\"5a921741\",\"components_sidetab.md\":\"adde93b1\",\"components_skeleton.md\":\"7dc8a7cf\",\"components_slider.md\":\"ac43ac89\",\"components_stepper.md\":\"8546a05b\",\"components_steps.md\":\"b7e4d00d\",\"components_sticky.md\":\"fc86eb2f\",\"components_stopwatch.md\":\"21112021\",\"components_swipecell.md\":\"656c0dd9\",\"components_swiper.md\":\"36cb4dbc\",\"components_switch.md\":\"4410d286\",\"components_tab.md\":\"403067fe\",\"components_tabbar.md\":\"f5582141\",\"components_tabview.md\":\"b06998e3\",\"components_tag.md\":\"fac4fb0d\",\"components_timeago.md\":\"c3f771bb\",\"components_timeline.md\":\"302d22f0\",\"components_toast.md\":\"11c92f1e\",\"components_index.md\":\"109baddd\",\"design_color.md\":\"fb90f8c4\",\"design_index.md\":\"8fb6e11d\",\"guide_dark.md\":\"1e1e9727\",\"guide_import.md\":\"efc436f4\",\"guide_locale.md\":\"abd89520\",\"guide_theme.md\":\"94717739\",\"index.md\":\"46b691ce\"}")</script>
    <script type="module" async src="/vfox/assets/app.6b4e8d9e.js"></script>
    
  </body>
</html>