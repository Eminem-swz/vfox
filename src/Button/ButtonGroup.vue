<template>
  <div
    class="fx-button-group"
    :class="[
      'size--' + options.size,
      'pattern--' + options.pattern,
      'shape--' + options.shape,
      'count--' + (children.length || 1)
    ]"
  >
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType, provide, reactive, watch } from 'vue'
import { createEnumsValidator, getEnumsValue } from '@/helpers/validator'
import { SIZE_TYPES } from '@/helpers/constants'
import { BUTTON_PATTERN_TYPES, BUTTON_SHAPE_TYPES } from '@/Button/button'
import type { SizeType } from '../helpers/types'
import type {
  ButtonPatternType,
  ButtonShapeType,
  ButtonGroupOptions
} from './types'
import { useGroup } from '@/hooks/use-group'

export default defineComponent({
  name: 'fx-button-group',
  props: {
    size: {
      type: String as PropType<SizeType>,
      validator: createEnumsValidator(SIZE_TYPES),
      default: null
    },
    pattern: {
      type: String as PropType<ButtonPatternType>,
      validator: createEnumsValidator(BUTTON_PATTERN_TYPES),
      default: null
    },
    shape: {
      type: String as PropType<ButtonShapeType>,
      validator: createEnumsValidator(BUTTON_SHAPE_TYPES),
      default: null
    }
  },
  setup(props) {
    const { children } = useGroup('button')

    const options = reactive<Partial<ButtonGroupOptions>>({})

    watch(
      props,
      val => {
        options.size = getEnumsValue(SIZE_TYPES, val.size)
        options.pattern = getEnumsValue(BUTTON_PATTERN_TYPES, val.pattern)
        options.shape = getEnumsValue(BUTTON_SHAPE_TYPES, val.shape)
      },
      {
        immediate: true
      }
    )

    provide('fxButtonGroupOptions', options)

    return { children, options }
  }
})
</script>
