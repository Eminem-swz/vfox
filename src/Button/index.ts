import { App } from 'vue'
import { SFCWithInstall } from '../utils/types'
import Button from './Button.vue'

const _Button: SFCWithInstall<typeof Button> = Object.assign(Button, {
  install: function(app: App) {
    app.component(Button.name, Button)
  }
})

export default _Button
