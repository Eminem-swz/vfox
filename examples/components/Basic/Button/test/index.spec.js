/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import Button from '@/Button'
import ButtonGroup from '@/ButtonGroup'

describe('Button', () => {
  test('should render default slot correctly', () => {
    const wrapper = shallowMount(Button, {})

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render add icon correctly', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        icon: 'EditOutlined'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('ButtonGroup', () => {
  test('should render 2 buttons slot correctly', () => {
    const wrapper = shallowMount(ButtonGroup, {
      slots: {
        default: [Button, Button]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
