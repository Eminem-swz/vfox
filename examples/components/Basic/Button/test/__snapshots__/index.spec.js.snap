// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render add icon correctly 1`] = `
<button type="button" class="fx-button type--default size--large pattern--default shape--rectangle has--icon">
  <icon-stub icon="EditOutlined" size="0" width="0" height="0"></icon-stub> <span>content</span>
</button>
`;

exports[`Button should render default slot correctly 1`] = `
<button type="button" class="fx-button type--default size--large pattern--default shape--rectangle">
  <!----> <span>content</span>
</button>
`;

exports[`ButtonGroup should render 2 buttons slot correctly 1`] = `
<div class="fx-button-group size--large pattern--default shape--rectangle count--1"><button type="button" class="fx-button type--default size--large pattern--default shape--rectangle">
    <!----> <span>按钮</span>
  </button><button type="button" class="fx-button type--default size--large pattern--default shape--rectangle">
    <!----> <span>按钮</span>
  </button></div>
`;
